worker/worker.go

```diff
@@ -867,14 +866,14 @@ func (w *worker) slabMigrateHandler(jc jape.Context) {
 		return
 	}
 
-	// fetch all contract set contracts
+	// update uploader contracts
 	ulContracts, err := w.bus.ContractSetContracts(ctx, up.ContractSet)
 	if jc.Check("couldn't fetch contracts from bus", err) != nil {
 		return
 	}
+	w.uploadManager.update(ulContracts, up.CurrentHeight)
```

Comment:
Yeah it recreates the set of uploaders but this is something I want to change actually. It's a bit tricky because it doesn't play well with contract sets. I think the upload manager should manage a list of uploaders per contract set.

Reply:
I improved it by updating the manager in `newUpload` and pruning contracts that expire. This way unknown are new contracts are added to the manager on the fly and old/expired ones get purged. This improves the behaviour when users pass random contract sets, something which the old upload manager didn't deal with.